composer/algorithms/agc/agc.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Core automatic gradient clipping classes and functions."""
+
+from __future__ import annotations
+
+import logging
+from typing import Optional
+
+import torch
+
+from composer.core import Algorithm, Event, State
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["AGC", "apply_agc"]
+
+
+def apply_agc(
+    model: torch.nn.Module,
+    clipping_threshold: float = 0.01,
+    eps: float = 1e-3,
```

Comment:
I'm not a fan of the `eps` argument, as we don't know ahead of time how the user is initializing their model weights. `1e-3` might be really big or really small.

Instead, can we jsut get rid of the `eps` entirely? We can remove all the clamps and epsilons except the very last one and I think it will be safe because `torch.clamp` replaces infinities:

```
>>> a = torch.ones(3) / torch.zeros(3)
>>> a
tensor([inf, inf, inf])
>>> torch.clamp(a, max=1)
tensor([1., 1., 1.])
```

Reply:
When I run that code I actually get nans which don't get clipped by the clamp call. Seems like we have to specify a min value somewhere? What do you think of keeping `grad_norm = _unitwise_norm(grad).clamp_(min=1e-6)`, but removing eps: `w_norm = _unitwise_norm(weights)`~~.clamp_(eps)~~, like I did here 
8ef0af4